metar/Metar.py

```diff
@@ -286,43 +290,56 @@ def xlate_loc( loc ):
 
 ## Helper functions
 
-def _report_match(handler,match):
+def _report_match(handler, match):
   """Report success or failure of the given handler function. (DEBUG)"""
   if match:
-      print(handler.__name__," matched '"+match+"'")
+      _logger.debug(handler.__name__ + " matched '" + match + "'")
   else:
-      print(handler.__name__," didn't match...")
-      
+      _logger.debug(handler.__name__ + " didn't match...")
+
 def _unparsedGroup( self, d ):
     """
     Handle otherwise unparseable main-body groups.
     """
     self._unparsed_groups.append(d['group'])
-      
+
 ## METAR report objects
 
 debug = False
 
 class Metar(object):
   """METAR (aviation meteorology report)"""
-  
+
   def __init__(self, metarcode, month=None, year=None, utcdelta=None,
-               strict=True):
+               strict=True, on_error='raise'):
```

Comment:
Will want to boggle this some more, how `strict` and `on_error` are named.  My initial thought is to be more explicit and name this `on_parse_error` or maybe `strict` should be extended to cover the parse errors too?

Reply:
I think the latter makes the most sense, but get complicated as:
`strict = True` -> raise errors
`strict= False` -> warn or log -- we need to pick a default

so maybe the logic is:
```python
if strict is True:
    raise ....
else:
    if strict == 'warn':
        warnings.warn(...)
    elif strict == 'log' or strict is False
        _logger.debug(....)
    else:
        raise ValueError("`strict` should be one of {True, 'log', 'warn'}")
        # don't want to encourage the ambiguity of False
```

Reply:
maybe, but your example logic still has a bug when `strict === False`

Reply:
Does it? I'm not doing any string methods in that example.

In any case, tinkering further, I'm thinking something more along these lines:


```python
        
def validate_strict(strict):
    _valid_strings = {'raise', 'log', 'warn'}
    _bool_map = {
        True: 'raise',
        False: 'log'
    }  
    if not isinstance(strict, bool) and not str(strict).lower() in _valid_strings:
        raise ValueError('bad value for strict')
    elif isinstance(strict, bool):
        return _bool_map[strict]
    else:
        return strict.lower()
```
That way, we can only worry about "raise," "log", and "warn"

And then this goes as I expect

```python
def test_strict_logic(strict):
    strict = validate_strict(strict)
    if strict == 'raise':
        x = 1
    elif strict == 'warn':
        x = 2
    elif strict == 'log':
        x = 3
    else: # unreachable
        x = 4
    return x

strict_values = [
    (True, 1), 
    ('raise', 1), 
    ('warn', 2), 
    (False, 3), 
    ('log', 3), 
    ('junk', 4), 
]
for strict, expected in strict_values:
    assert expected == test_strict_logic(strict)
```

but perhaps I'm over thinking this